ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -54,8 +58,12 @@ internal interface PaywallViewModel {
      */
     fun purchaseSelectedPackage(activity: Activity?)
 
+    suspend fun awaitPurchaseSelectedPackage(activity: Activity?)
+
     fun restorePurchases()
 
+    suspend fun awaitRestorePurchases()
```

Comment:
Are these (`await*()`) functions added only for tests? If so, I don't think we need that. See next comment 👇

Reply:
Yes, they were added specifically for tests. If I'd been aware of that testing flow before I'd refactored I probably wouldn't have bothered. But now it's done, I think it makes for cleaner test code, and having await variants available doesn't seem like a bad thing to me generally. So at this point, I'd be inclined to leave it as-is, but I will revert if you'd prefer it as it was before. Let me know!

Reply:
if these methods are public and only for tests then it'd be great to find a way to do without them

Reply:
I think these are internal. The class is `internal class PaywallViewModelImpl` and these functions are part of the interface `internal interface PaywallViewModel`. It also looks like I could take them out of the interface and then mark them as `internal` right in the class.

Reply:
They are internal indeed because the class is internal, but they are "public" in the sense of this class' api. 

I'd really prefer to not add any code specifically for tests, regardless of visibility. A test should verify actual behavior. When adding code specifically for tests, there's a risk that actual behavior and tested behavior deviates. Maybe not now, but that risk grows after more time passes and more changes are made to this class. For instance, we need to remember to copy any change in `purchaseSelectedPackage()` to `awaitPurchaseSelectedPackage()`. That's very error prone imo. 

I don't think we'd need to revert entirely: 
1. The tests can be adjusted to not call the `await*()` variants, as suggested below.
2. `awaitPurchaseSelectedPackage()` can then be removed entirely from both the interface and the class.
3. `awaitRestorePurchases()` can then be removed from the interface, and be made private in the class.

Happy to discuss more of course!

Reply:
Also, I see your point that `await*()` can be cleaner, I think there's an argument to be made for that. However, then I'd argue that the _fire-and-forget_ methods should be removed (`purchaseSelectedPackage()` and `restorePurchases()`). 
  
But this is a more risky change since we'd need to get a CoroutineScope in Compose ([`rememberCoroutineScope()`](https://developer.android.com/develop/ui/compose/side-effects#remembercoroutinescope)), which has a different life cycle from a `viewModelScope`. So we need to verify that the behavior is still the same. For instance, we need to make sure that that scope doesn't get cancelled when the Activity pauses (e.g. when it is overlaid by the purchase flow).